lib/common/puyo/stepper.js

```diff
@@ -264,10 +268,19 @@ class DuelStepper extends BasicStepper {
     const hasStepped = {};
     const nuisanceReceivers = [];
     const eventsForPlayer = this.distributeEvents(state, events);
-
     let gameOver = false;
+    let terminated = false;
+    let terminationReason;
+    let terminatedPlayer;
+
     ++state.time;
 
+    if (events.find(event => event.type === 'terminated')) {
+      terminated = true;
+      terminationReason = event.reason;
```

Comment:
Wow. I think I missed a step there. I'll store the potential event in a variable.